pallets/settlement/src/lib.rs

```diff
@@ -1209,74 +1205,96 @@ impl<T: Config> Module<T> {
         caller_did: IdentityId,
         weight_meter: &mut WeightMeter,
     ) -> DispatchResult {
-        // Verifies that there are no pending affirmations for the given instruction
-        ensure!(
-            Self::instruction_affirms_pending(instruction_id) == 0,
-            Error::<T>::NotAllAffirmationsHaveBeenReceived
-        );
-        // Verifies that all mediator's affirmations are still valid
-        Self::ensure_non_expired_affirmations(&instruction_id)?;
-
-        // Ensures the instruction is pending or has failed at least one time
-        let instruction_status = Self::instruction_status(instruction_id);
-        ensure!(
-            instruction_status == InstructionStatus::Pending
-                || instruction_status == InstructionStatus::Failed,
-            Error::<T>::InvalidInstructionStatusForExecution
-        );
-
-        let venue_id = Self::instruction_details(instruction_id).venue_id;
-
-        // NB: The order of execution of the legs matter in some edge cases around compliance.
-        // E.g: Consider a token with a total supply of 100 and maximum percentage ownership of 10%.
-        // In a given moment, Alice owns 10 tokens, Bob owns 5 and Charlie owns 0.
-        // Now, consider one instruction with two legs: 1. Alice transfers 5 tokens to Charlie; 2. Bob transfers 5 tokens to Alice;
-        // If the second leg gets executed before the first leg, Alice will momentarily hold 15% of the asset and hence the settlement will fail compliance.
-        let mut instruction_legs: Vec<(LegId, Leg)> =
-            InstructionLegs::iter_prefix(&instruction_id).collect();
-        instruction_legs.sort_by_key(|leg_id_leg| leg_id_leg.0);
-
-        let instruction_asset_count = AssetCount::from_legs(&instruction_legs);
-        weight_meter
-            .check_accrue(<T as Config>::WeightInfo::execute_instruction_paused(
-                instruction_asset_count.fungible(),
-                instruction_asset_count.non_fungible(),
-                instruction_asset_count.off_chain(),
-            ))
-            .map_err(|_| Error::<T>::WeightLimitExceeded)?;
+        let mut failed_leg_id = None;
+        let tx_result = with_transaction(|| {
+            // Ensures the number of pending affirmations is zero
+            let n_pending_affirmations = InstructionAffirmsPending::take(instruction_id);
```

Comment:
Is there any known way for `InstructionAffirmsPending` to be 0, but for `Self::ensure_no_missing_affirmation` to fail?

Reply:
It shouldn't be possible of that to happen, unless a bug is introduced later.  All of the storage for affirmations needs to be cleared anyways, so it isn't much more expensive to check the values.